[RSDK-3436] Update slam docs with tuning spike info (#1410)
viamrobotics/docs: docs/services/slam/cartographer/_index.md

| Parameter Mode | Description | Inclusion | Default Value | Notes |
| -------------- | ----------- | --------- | ------------- | ----- |
| `mode` | `2d` | **Required** | None | |

> | `optimize_every_n_nodes` | How many trajectory nodes are inserted before the global optimization is run. | Optional | `3` | Setting to `0` will disable global SLAM, using local SLAM only. |

Reviewer: ```suggestion
| `optimize_every_n_nodes` | How many trajectory nodes are inserted before the global optimization is run. | Optional | `3` | To disable global SLAM and use only local SLAM, set this to `0`. |
```

> | `num_range_data` | Number of measurements in each submap. | Optional | `30` | |

Author: Yes will do!

> | `num_range_data` | Number of measurements in each submap. | Optional | `30` | |

Reviewer 2: ty!

> | `missing_data_ray_length` | Replaces the length of ranges that are further than `max_range` with this value. | Optional | `25` | Should be the same as `max_range`. |

Author: From what I've seen there might be some applications where it's worth setting to a bit lower than max_range. It might be better to change it to "Typically the same as max_range" to indicate this?

> | `missing_data_ray_length` | Replaces the length of ranges that are further than `max_range` with this value. | Optional | `25` | Should be the same as `max_range`. |

Reviewer 2: That phrasing sounds good to me!

| `max_range` | Maximum range of valid measurements. | Optional | `25` | For an RPlidar A3, set this value to `25`. For an RPlidar A1, use `12`. |
| `min_range` | Minimum range of valid measurements. | Optional | `0.2` | For an RPlidar A3, set this value to `0.2`. For RPlidar A1, use `0.15`. |
| `max_submaps_to_keep` | Number of submaps to use and track for localization. | Optional | `3` | Only for [LOCALIZING mode](#mapping-modes). |
| `fresh_submaps_count` | Length of submap history considered when running SLAM in updating mode. | Optional | `3` | Only for [UPDATING mode](#mapping-modes). |
| `min_covered_area` | The minimum overlapping area, in square meters, for an old submap to be considered for deletion. | Optional | `1.0` | Only for [UPDATING mode](#mapping-modes). |
| `min_added_submaps_count` | The minimum number of added submaps before deletion of the old submap is considered. | Optional | `1` | Only for [UPDATING mode](#mapping-modes). |

> | `occupied_space_weight` | Emphasis to put on scanned data points between measurements. Higher values make it harder to overwrite prior scanned points. | Optional | `20.0` | Weights are all relative. |

Reviewer 3: [q] should the second sentence be placed in the notes column of the table?

> | `occupied_space_weight` | Emphasis to put on scanned data points between measurements. Higher values make it harder to overwrite prior scanned points. | Optional | `20.0` | Weights are all relative. |

Author: The three weights are relative to each other

> | `occupied_space_weight` | Emphasis to put on scanned data points between measurements. Higher values make it harder to overwrite prior scanned points. | Optional | `20.0` | Weights are all relative. |

Reviewer 2: Got it, can we say something like "relative to translation_weight and rotation_weight" and similar in the rows for translation_weight and rotation_weight?

> | `occupied_space_weight` | Emphasis to put on scanned data points between measurements. Higher values make it harder to overwrite prior scanned points. | Optional | `20.0` | Weights are all relative. |

Author: Yes will do!

| `translation_weight` | Emphasis to put on expected translational change from pose extrapolator data between measurements. Higher values make it harder for scan matching to translate prior points. | Optional | `10.0` | Weights are all relative. |

> | `rotation_weight` | Emphasis to put on expected rotational change from pose extrapolator data between measurements. Higher values make it harder for scan matching to rotate prior points. | Optional | `1.0` | Weights are all relative. |

Author: Rotating an individual scan- I'll change that to be more clear
